crypto/err/err.c

```diff
@@ -895,3 +895,21 @@ void err_clear_last_constant_time(int clear)
                                      0, ERR_FLAG_CLEAR);
     es->err_flags[top] |= clear;
 }
+
+void ossl_err_raise_malloc_failure(const char *file, int line)
+{
+    ERR_STATE *es;
+
+    /* prevent potential malloc error loop while reporting malloc eror */
+    if (!RUN_ONCE(&err_init, err_do_init))
+        return;
+    es = CRYPTO_THREAD_get_local(&err_thread_local);
```

Comment:
If there is no error stack, then can't you assume no malloc failures have been reported?

Reply:
The problem is that the malloc failure could be caused by the error stack allocation.
However it also means the malloc failure error won't be raised if it is the first ever error raised (and there was no other ERR_ call done before).

Reply:
> The problem is that the malloc failure could be caused by the error stack allocation. However it also means the malloc failure error won't be raised if it is the first ever error raised (and there was no other ERR_ call done before).

First malloc failure *in the current thread* tho.  A comment should be added explaining that.  This PR is probably better, but it's clearly not complete.

Reply:
Also here, I've added a comment pointing out the limitation.